[ENH] Bep 005: Arterial Spin Labeling (#669)
bids-standard/bids-specification: src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

|------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | EchoTime               | RECOMMENDED, but REQUIRED if corresponding fieldmap data is present, or the data comes from a multi echo sequence or Arterial Spin Labeling                 | [array][] of [numbers][] | The echo time (TE) for the acquisition, specified in seconds. A single value applies to all volumes, or an array provides a value for each volume. Corresponds to DICOM Tag 0018, 0081 `Echo Time` (please note that the DICOM term is in milliseconds not seconds). 4D files with variable TE should only be used when volume timing is critical for interpretation of the data, as in ASL or variable TE fMRI sequences.                                                                                                                                                                                                                                                                                                                                                                                                                      |

Reviewer: Apologies for the naive question, but is multi-echo ASL possible? I came across [Paschoal et al. (2019)](https://www.hindawi.com/journals/cmra/2019/5040465/), which seems to compare a dual-echo PCASL sequence to a PCASL sequence with varying echoes.

Reviewer: I ask because, if multi-echo ASL is possible, I think it's important that echoes from the same excitation be separated into different files with the `echo` entity, rather than concatenated as they might be in a variable echo sequence.

Reviewer 2: I'll drop this question in the group!

Reviewer 2: @tsalo Hi Taylor, 

Dual-Echo ASL/fMRI does indeed exist but is not used a lot, which is why we chose not to support  this in our current first ASL BIDS version. This specific case would still fit in the current specification, as the first echo would be stored in /perf/ and the second in /func/, for other multi-echo ASL sequences - e.g. to measure the BBB - we will wait with integrating them. The new EchoTime with options both “number or array of numbers” will be helpful for “single-echo” ASL sequences where the echo time varies over volumes (e.g. QUASAR). Hope this clarifies things?

Reviewer: Thank you, that is very helpful. Support can wait until the future, but I'm just glad to hear that the plan isn't to concatenate the separate echoes within a single file.

|------------------|-----------------------------------------------------------------------------------------|----------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | FlipAngle        | RECOMMENDED, but REQUIRED for Arterial Spin Labeling in case `LookLocker` is set `true` | [number][] or [array][] of [numbers][] | Flip angle (FA) for the acquisition, specified in degrees. Corresponds to: DICOM Tag 0018, 1314 `Flip Angle`. The data type number may apply to files from any MRI modality concerned with a single value for this field, or to the files in a [file collection](../99-appendices/10-file-collections.md) where the value of this field is iterated using the [flip entity](../99-appendices/09-entities.md#flip). The data type array provides a value for each volume in a 4D dataset and should only be used when the volume timing is critical for interpretation of the data, such as in [ASL](#arterial-spin-labeling-perfusion-data) or variable flip angle fMRI sequences. |

Reviewer 2: @effigies I think you are right about this. Will add @HenkMutsaerts to confirm.

> | FlipAngle        | RECOMMENDED, but REQUIRED for Arterial Spin Labeling in case `LookLocker` is set `true` | [number][] or [array][] of [numbers][] | Flip angle (FA) for the acquisition, specified in degrees. Corresponds to: DICOM Tag 0018, 1314 `Flip Angle`. The data type number may apply to files from any MRI modality concerned with a single value for this field, or to the files in a [file collection](../99-appendices/10-file-collections.md) where the value of this field is iterated using the [flip entity](../99-appendices/09-entities.md#flip). The data type array provides a value for each volume in a 4D dataset and should only be used when the volume timing is critical for interpretation of the data, such as in [ASL](#arterial-spin-labeling-perfusion-data) or variable flip angle fMRI sequences. |

Reviewer 3: Yes good suggestion by Chris

}
```

## Arterial Spin Labeling perfusion data

Template:

```Text
sub-<label>/[ses-<label>/]
    perf/
       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_asl.nii[.gz]
       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_asl.json
       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_aslcontext.tsv
       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_m0scan.nii[.gz]
       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_m0scan.json
       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_run-<index>]_labeling.jpg
```

The complete ASL time series should be stored as a 4D NIfTI file in the original acquisition order, accompanied by two ancillary 
files: `*_asl.json` and `*_aslcontext.tsv`. The `*_asl.nii.gz` and `*_m0scan.nii.gz` should contain appropriately scaled data and no scaling factors 

> should be stored in the ancillary files or in the NIfTI header. 

Reviewer 4: I still have the same comment: https://github.com/bids-standard/bids-specification/pull/652/files#r520070205

Reviewer 2: @gllmflndn I changed it now!

Reviewer 4: Thank you!

> | M0                                | REQUIRED                                                                                                                        | [number][] or [string][] or [boolean][] | Describes the available M0 information or the location of an `m0scan` file, as either: Numerical: a single numerical whole-brain M0 value, for example obtained from CSF in another sequence; SeparateFile: `*_m0scan.nii[.gz]`; Within ASL time-series: Boolean: `true` referring to an `m0scan` within the time-series (as defined within the `*_aslcontext.tsv`); `false` referring to the use of the (average) `control` image for M0 calibration.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |

Reviewer: Is the `asl` file's `M0` not redundant with the `m0scan`'s `IntendedFor`?

Reviewer 2: Hi @tsalo, we are rethinking the M0 field, based on an earlier comment. This redundancy will probably be solved with the new approach. I'll add it once we finished it and link to you.

Reviewer 2: @tsalo We updated it using the suggestions. Feel free to take a look.

### `m0scan` metadata fields

Some common metadata fields are required for the `*_m0scan.json`: `RepetitionTime` or `VolumeTiming`.

| **Key name**         | **Requirement level** | **Data type**                          | **Description**                                                                                                                                                                                                                                                                                                                                                                   |
|----------------------|-----------------------|----------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | IntendedFor          | REQUIRED              | [string][] or [array][] of [strings][] | One or more filenames with paths relative to the subject subfolder, with forward slashes, referring to ASL time series for which the m0scan is intended.AcquisitionVoxelSize                                                                                                                                                                                                      |

Reviewer 2: @effigies Same for IntendedFor. It has been used already in the specification. Now it has a paragraph and definition on the bottom of the page (Fieldmap Case 4). I can just refer to that one, I assume? Or should that paragraph (from fieldmap Case 4) go to common?

> | AcquisitionVoxelSize              | RECOMMENDED                                                                                                                     | [array][] of [numbers][]                | An array of numbers with a length of 3, in millimeters. This parameter denotes the original acquisition voxel size, excluding any inter-slice gaps and before any interpolation or resampling within reconstruction or image processing. Any point spread function effects, for example due to T2-blurring, that would decrease the effective resolution are not considered here.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |

Reviewer 2: @effigies While I am working on common fields: should this one become a common field as well?

The following table recapitulates the ASL field dependencies. If Source field (column 1) contains the Value specified in column 2, then the Requirements in column 4 are imposed on the Dependent fields in column 3.

| **Source field**         | **Value**    | **Dependent field**  | **Requirements**                             |
|--------------------------|--------------|----------------------|----------------------------------------------|
| MRAcquisitionType        | 2D / 3D      | SliceTiming          | \[X\] / \[\]                                 |
| LookLocker               | true         | FlipAngle            | \[X\]                                        |
| ArterialSpinLabelingType | PASL         | BolusCutOffFlag      | \[X\]                                        |
| BolusCutOffFlag          | true / false | BolusCutOffDelayTime | \[X\] / \[\]                                 |
| BolusCutOffFlag          | true / false | BolusCutOffTechnique | \[X\] / \[\]                                 |

> | M0Type                   | Separate     | */perf/              | contains *_m0scan.nii[.gz] and *_m0scan.json |

Reviewer 2: @sappelhoff Hi Stefan
I am using *_m0scan and likewise namings in the dependency table. However, I am wondering if this could screw up the table for read the docs (like it did for the [X])? I can not check this, as checks are failing now on these hyperlinks. Do you know if I should add \ here as well?

Author: Yes, all Special markdown characters have to be escaped using a backslash `\`,  or by putting them into an inline code block using Backtricks: `

Special markdown characters are * _ |  and so on. I usually prefer inline code blocks, but I recommend to stay consistent with how it is done elsewhere in the specification

Author: checks are working now so you should be able to verify @patsycle

| BackgroundSuppressionPulseTime    | OPTIONAL, RECOMMENDED if `BackgroundSuppression` is `true`                    | [array][] of [numbers][]                | Array of numbers containing timing, in seconds, of the background suppression pulses with respect to the start of the labeling. In case of multi-PLD with different background suppression pulse times, only the pulse time of the first PLD should be defined.                                                                                                                                                                                                                                                                                                                                                                                                                                                 |

> | VascularCrushingVENC              | OPTIONAL, RECOMMENDED if `VascularCrushing` is `true`                         | [number][] or \[array][] of [numbers][] | The crusher gradient strength, in centimeters per second. Specify either one number for the total time-series, or provide an array of numbers, for example when using QUASAR, using the value zero to identify volumes for which `VascularCrushing` was turned off. Corresponds to DICOM Tag 0018,925A `ASL Crusher Flow Limit`.                                                                                                                                                                                                                                                                                                                                                                                |

Author: Here you wrote `\[array][]`, but the leading backslash prevents a proper link rendering, see this screenshot:

![grafik](https://user-images.githubusercontent.com/9084751/102772111-44de5700-4387-11eb-9036-3a738795faa0.png)

you just need to remove the backslash, and adjust the table pipes.

Reviewer 2: @sappelhoff This should be fixed now! Thanks for spotting it.

}
```

## Arterial Spin Labeling perfusion data

Template:

```Text
sub-<label>/[ses-<label>/]
    perf/
       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_asl.nii[.gz]
       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_asl.json
       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_aslcontext.tsv
       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_m0scan.nii[.gz]
       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_m0scan.json

>        sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_run-<index>]_labeling.jpg

Reviewer 2: @effigies Task is possible and we were considering adding it. However, there might be more specifics to think about when adding task-asl. So we decided not to include this yet and to stick with the white paper asl approaches. Task-asl will probably be added in a later version.

|-----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| control         | The control image is acquired in the exact same way as the label image, except that the magnetization of the blood flowing into the imaging region has not been inverted.              |
| label           | The label image is acquired in the exact same way as the control image, except that the blood magnetization flowing into the imaging region has been inverted.                         |
| m0scan          | The M0 image is a calibration image, used to estimate the equilibrium magnetization of blood.                                                                                          |
| deltam          | The deltaM image is a perfusion-weighted image, obtained by the subtraction of `control` - `label`.                                                                                    |

> | cbf             | The cerebral blood flow (CBF) image is produced by dividing the deltaM by the M0, quantified into `mL/100g/min` (See also [doi:10.1002/mrm.25197](https://doi.org/10.1002/mrm.25197)). |

Reviewer 5: Note that if this change is adopted, it will cascade to other cases. Happy to review for consistency.

> | cbf             | The cerebral blood flow (CBF) image is produced by dividing the deltaM by the M0, quantified into `mL/100g/min` (See also [doi:10.1002/mrm.25197](https://doi.org/10.1002/mrm.25197)). |

Reviewer 2: Hi @effigies , initially we used capitals, but we changed it to this, based on the rest of the specification. It appears that 'not using capitals' is the standard within the specification until now. Also, I suppose this leads to less error in coding.

> | cbf             | The cerebral blood flow (CBF) image is produced by dividing the deltaM by the M0, quantified into `mL/100g/min` (See also [doi:10.1002/mrm.25197](https://doi.org/10.1002/mrm.25197)). |

Reviewer 3: I agree with the readability, as long as this part is consistent with the rest of BIDS; up to the BIDS maintainers :)

> | cbf             | The cerebral blood flow (CBF) image is produced by dividing the deltaM by the M0, quantified into `mL/100g/min` (See also [doi:10.1002/mrm.25197](https://doi.org/10.1002/mrm.25197)). |

Reviewer 2: So  @effigies @sappelhoff, what is the conclusion?

> | cbf             | The cerebral blood flow (CBF) image is produced by dividing the deltaM by the M0, quantified into `mL/100g/min` (See also [doi:10.1002/mrm.25197](https://doi.org/10.1002/mrm.25197)). |

Author: We actually have a recommendation on that here: https://bids-specification.readthedocs.io/en/latest/02-common-principles.html#tabular-files

> It is RECOMMENDED that the column names in the header of the TSV file are written in snake_case with the first letter in lower case (for example, variable_name, not Variable_name).

I completely forgot about that at first ... but since it's there, we should adhere to it. Lower case as it is now would be fine with me - but let's hear Chris' comment.

> | cbf             | The cerebral blood flow (CBF) image is produced by dividing the deltaM by the M0, quantified into `mL/100g/min` (See also [doi:10.1002/mrm.25197](https://doi.org/10.1002/mrm.25197)). |

Reviewer 5: These are column contents, not names, right? Or do they need to be column names elsewhere?

> | cbf             | The cerebral blood flow (CBF) image is produced by dividing the deltaM by the M0, quantified into `mL/100g/min` (See also [doi:10.1002/mrm.25197](https://doi.org/10.1002/mrm.25197)). |

Reviewer 2: @effigies They are the volume_names used in the aslcontext.tsv.

> | cbf             | The cerebral blood flow (CBF) image is produced by dividing the deltaM by the M0, quantified into `mL/100g/min` (See also [doi:10.1002/mrm.25197](https://doi.org/10.1002/mrm.25197)). |

Reviewer 5: I don't believe there's any precedent for requiring all TSV contents to be lowercase or snake case.

This really doesn't matter, though. Up to you.

> | cbf             | The cerebral blood flow (CBF) image is produced by dividing the deltaM by the M0, quantified into `mL/100g/min` (See also [doi:10.1002/mrm.25197](https://doi.org/10.1002/mrm.25197)). |

Reviewer 2: @effigies In that case, I think I'd keep it as it is.

If the `control` and `label` images are not available, their derivative `deltam` should be stored within the `*_asl.nii[.gz]` and specified in the `*_aslcontext.tsv` instead. If 
the `deltam` is not available, `cbf` should be stored within the `*_asl.nii[.gz]` and specified in the `*_aslcontext.tsv`. When `cbf` is stored within the `*_asl.nii[.gz]`, its 
units need to be specified in the `*_asl.json` as well. See below examples of these cases, in order of decreasing preference.

#### Case 1: `*_asl.nii[.gz]` consists of volume_types `control`, `label`

In most cases, the ASL timeseries, provided by the scanner, consist of a series of `control` and `label`, and optionally `m0scan`, volumes. In this case, only the `control`, 
`label`, and optionally `m0scan` volumes should be stored in the `*_asl.json`, and the exact volume_type series should be specified in the `*_aslcontext.tsv`. The optional 

> `deltam` or `cbf` volumes should be stored and specified as derivative. Example of `*_aslcontext.tsv`:

Reviewer 2: @effigies Currenlty, there is no specification for derivatives for ASL yet. Adding this would require another round of discussion on which info should be provided for those derivatives. So at this stage, storing these optional DeltaM and CBF images in derivatives is not implemented. This is also planned for a later version. I am not sure how to specify this here?

> `deltam` or `cbf` volumes should be stored and specified as derivative. Example of `*_aslcontext.tsv`:

Reviewer 5: Then could you clarify what you mean by "The optional `deltam` or `cbf` volumes should be stored and specified as derivative."? What I had above was my best guess as to what I thought it meant.

> `deltam` or `cbf` volumes should be stored and specified as derivative. Example of `*_aslcontext.tsv`:

Reviewer 2: @HenkMutsaerts We should discuss this!

> `deltam` or `cbf` volumes should be stored and specified as derivative. Example of `*_aslcontext.tsv`:

Reviewer 2: @effigies I will update by deleting 'The optional deltam or cbf volumes should be stored and specified as derivative' in these cases. The 3 cases will also move to the appendix. Feel free to take a look after change.

```Text
volume_type
control
label
control
label
m0scan
```

#### Case 2: `*_asl.nii[.gz]` consists of volume_types `deltam` (scanner does not export `control` or `label` volumes)

In some cases, `control` and `label` volumes are lacking within the acquired ASL timeseries, but the intermediate `deltam` - and optionally a `m0scan` - volume is 
reconstructed/exported by the scanner. In this case, the `deltam` should be included in the `*_asl.nii[.gz]` and specified in the `*_aslcontext.tsv`. An optional `cbf` volume 

> should be stored and specified as derivative. Example of `*_aslcontext.tsv`:

Reviewer 2: @effigies Thanks for the suggestions. We were wondering if it wouldn't be cleaner to add these cases in the ASL appendix which I am creating now for the summary images and flowchart? What do you think and the other maintainers think? @sappelhoff

> should be stored and specified as derivative. Example of `*_aslcontext.tsv`:

Author: I think it fits here but wouldn't oppose you moving it to the appendix if you feel strongly about it.

```Text
volume_type
deltam
m0scan
```

#### Case 3: `*_asl.nii[.gz]` consists of volume_type `cbf` (scanner does not export `control`, `label`, or `deltaM` volumes) 

> If `control` and `label` or intermediate ASL volumes are not reconstructed or exported, but a pre-calculated `cbf` - and optionally a `m0scan` - volume is provided by the
> scanner, the `cbf` should be included in the `*_asl.nii[.gz]` and specified in the `*_aslcontext.tsv`. Example of `*_aslcontext.tsv`:

Reviewer 5: ```suggestion
If `control` and `label` or intermediate ASL volumes are not reconstructed or exported,
but a pre-calculated `cbf` - and optionally a `m0scan` - volume is provided by the scanner,
the `cbf` should be included in the `*_asl.nii[.gz]` and specified in the `*_aslcontext.tsv`.
Example of `*_aslcontext.tsv`:
```

```Text
volume_type
cbf
m0scan
```

### Scaling

> The `*_asl.nii.gz` and `*_m0scan.nii.gz` should contain appropriately scaled data, and no additional scaling factors are allowed other than the scale slope in the respective 
> NIfTI headers.

Reviewer 5: I think a bit more context for this statement might help. Here's some text that reflects my understanding:

```suggestion
ASL sequences may record in scale factors (slope and intercept) that affect
the interpretation of the DICOM data and which can potentially vary by time point.
NIfTI defines `scl_slope` and `scl_inter` scale factors that apply to the entire image.
The `*_asl.nii.gz` and `*_m0scan.nii.gz` images should contain appropriately scaled data
when applying NIfTI scale factors.
To avoid ambiguity, additional scale factors should not be defined in JSON sidecar files.
```

Please treat this as a suggestion and feel free to reject or further rework.

Reviewer 2: @effigies Thanks for the suggestion. I am adding @HenkMutsaerts here, as he has more insight in the scaling issues. Henk, what do you think about the suggestion?

### M0

> The `m0scan` can either be stored inside the 4D ASL time-series NIfTI file or as a separate NIfTI file, depending on whether it 
> was acquired within the ASL time-series or as a separate scan. These and other M0 options are specified in the required `M0Type` field of the `*_asl.json` file. 
> It can also be stored under `fmap/sub-<label>[_ses-<label>][_acq-<label>][_ce-<label>]_dir-<label>[_run-<index>]_m0scan.nii[.gz]`, 
> when the [pepolar approach](https://bids-specification.readthedocs.io/en/latest/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#case-4-multiple-phase-encoded-directions-pepolar) is used.

Reviewer 5: ```suggestion
The `m0scan` can either be stored inside the 4D ASL time-series NIfTI file
or as a separate NIfTI file,
depending on whether it was acquired within the ASL time-series or as a separate scan.
These and other M0 options are specified in the REQUIRED `M0Type` field of the `*_asl.json` file. 
It can also be stored under `fmap/sub-<label>[_ses-<label>][_acq-<label>][_ce-<label>]_dir-<label>[_run-<index>]_m0scan.nii[.gz]`, 
when the [pepolar approach](#case-4-multiple-phase-encoded-directions-pepolar) is used.
```

### `*_asl.json` file

> Depending on the method used for ASL acquisition ((P)CASL or PASL) different metadata fields are applicable. Additionally, some common metadata fields are required for the 
> `*_asl.json`: `MagneticFieldStrength`, `MRAcquisitionType`, `EchoTime`, `SliceTiming` in case `MRAcquisitionType` is defined as 2D, `RepetitionTimePreparation`, and 
> `FlipAngle` in case `LookLocker` is `true`.

Reviewer 5: ```suggestion
Depending on the method used for ASL acquisition ((P)CASL or PASL)
different metadata fields are applicable.
Additionally, some common metadata fields are required for the `*_asl.json`:
`MagneticFieldStrength`, `MRAcquisitionType`, `EchoTime`,
`SliceTiming` in case `MRAcquisitionType` is defined as 2D,
`RepetitionTimePreparation`, and `FlipAngle` in case `LookLocker` is `true`.
```

| LabelingDuration             | REQUIRED              | [number][] or [array][] of [numbers][] | Total duration of the labeling pulse train, in seconds, corresponding to the temporal width of the labeling bolus for `(P)CASL`. Specify either one number for total time-series or provide an array of numbers of different values for each volume in case of sequential acquisitions with varying labeling duration. In the case that an array of numbers is provided, its length should be equal to the number of volumes specified in `*_aslcontext.tsv`. An `m0scan` acquired within the ASL time-series, gets a value of zero. Corresponds to DICOM Tag 0018,9258 `ASL Pulse Train Duration`. |

> | PCASLType                    | RECOMMENDED           | [string][]                             | Type the gradient pulses used in the `control` condition: `balanced` or `unbalanced`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
> | CASLType                     | RECOMMENDED           | [string][]                             | Describes if a separate coil is used for labeling: `single-coil` or `double-coil`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |

Reviewer 5: Maybe?

```suggestion
| PCASLType                    | RECOMMENDED if ArterialSpinLabelingType is PCASL | [string][]                             | Type the gradient pulses used in the `control` condition: `balanced` or `unbalanced`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
| CASLType                     | RECOMMENDED if ArterialSpinLabelingType is CASL  | [string][]                             | Describes if a separate coil is used for labeling: `single-coil` or `double-coil`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |
```

Reviewer 2: @effigies sound logical

| BolusCutOffDelayTime  | OPTIONAL, REQUIRED if `BolusCutOffFlag` is `true` | [number][] or [array][] of [numbers][] | Duration between the end of the labeling and the start of the bolus cut-off saturation pulse(s), in seconds. This can be a number or array of numbers, of which the values must be non-negative and monotonically increasing, depending on the number of bolus cut-off saturation pulses. For Q2TIPS, only the values for the first and last bolus cut-off saturation pulses are provided. Based on DICOM Tag 0018,925F `ASL Bolus Cut-off Delay Time`. |
| BolusCutOffTechnique  | OPTIONAL, REQUIRED if `BolusCutOffFlag` is `true` | [string][]                             | Name of the technique used, for example Q2TIPS, QUIPSS, QUIPSSII. Corresponds to DICOM Tag 0018,925E `ASL Bolus Cut-off Technique`.                                                                                                                                                                                                                                                                                                                     |

### `m0scan` metadata fields

> The common metadata field `RepetitionTimePreparation` is required for the `*_m0scan.json`.

Reviewer 5: ```suggestion
The common metadata field `RepetitionTimePreparation` is REQUIRED for the `*_m0scan.json`.
```

| BackgroundSuppressionPulseTime    | OPTIONAL, RECOMMENDED if `BackgroundSuppression` is `true`                    | [array][] of [numbers][]               | Array of numbers containing timing, in seconds, of the background suppression pulses with respect to the start of the labeling. In case of multi-PLD with different background suppression pulse times, only the pulse time of the first PLD should be defined.                                                                                                                                                                                                                                                                                                                                                                                                                                                 |

> | VascularCrushingVenc              | OPTIONAL, RECOMMENDED if `VascularCrushing` is `true`                         | [number][] or [array][] of [numbers][] | The crusher gradient strength, in centimeters per second. Specify either one number for the total time-series, or provide an array of numbers, for example when using QUASAR, using the value zero to identify volumes for which `VascularCrushing` was turned off. Corresponds to DICOM Tag 0018,925A `ASL Crusher Flow Limit`.                                                                                                                                                                                                                                                                                                                                                                                |

Reviewer 2: @effigies No you are right! This discussion came up again, but we forgot about your previous comments. You are completely right, so I will change to VENC again. Thanks and sorry for the confusion!
